packages/terra-form-validation/docs/ValidationEvents.md

```diff
@@ -0,0 +1,103 @@
+# Validation Events
+
+The default behavior of *react-final-form* is to validate dynamically (while the user is interacting with the form, and on form load). To use this behavior, simply use the error key of the meta data passed to an individual form component, and pass it to the appropriate InputField for TextareaField, like so below.
+
+```jsx
+  <Field
+    name="description"
+  >
+    {({ input, meta, placeholder, ...rest }) => (
+      <InputField
+        {...rest}
+        label="Description"
+        error={meta.error}
+        isInvalid={!meta.valid}
+        required
+        onChange={(e) => {input.onChange(e.target.value);}}
+        value={input.value}
+        inputAttrs={{
+          type: 'text',
+          ...input,
+        }}
+      />
+    )}
+  </Field>
+```
+
+## Validation on submit
+
+Validating submit is easy as well. In addition to meta.error, use the boolean meta.submitFailed to identify when a form submission did not succeed.
+
+```jsx
+  <Field
+    name="description"
+  >
+    {({ input, meta, placeholder, ...rest }) => (
+      <InputField
+        {...rest}
+        label="Description"
+        error={meta.error}
+        isInvalid={meta.submitFailed && meta.error}
+        required
+        onChange={(e) => {input.onChange(e.target.value);}}
+        value={input.value}
+        inputAttrs={{
+          type: 'text',
+          ...input,
+        }}
+      />
+    )}
+  </Field>
+```
+
+## Disable form submit for an invalid form.
+
+The form render function contains attributes such as invalid and pristine (untouched state) that allow for you to enable and disable the submission button. All you need to do is pass in the attribute you wish to check in order to see if the button should be enabled or not.
```

Comment:
I feel form render function needs to be introduced before mentioned possible attributes to disable a button. How does this value get determined? Does this need to be handled by the consumer writing the renderForm function or where does these options become available? I feel we need a more detailed 'out-of-the-box' working example people can start with and then modify to fit their needs

Reply:
I've added more on this inside the main index page of this package. I detailed how the render function of both Form and Field work, and the most essential props do.